lib/recurly.d.ts

```diff
@@ -0,0 +1,785 @@
+  quantity?: number;
+  unitAmount?: number;
+  createdAt?: Date;
+  updatedAt?: Date;
+  expiredAt?: Date;
+}
+
+export interface AddOnMini {
+  id?: string;
+  object?: string;
+  code?: string;
+  name?: string;
+  accountingCode?: string;
+}
+
+export interface UniqueCouponCode {
+  id?: string;
+  object?: string;
+  code?: string;
+  state?: string;
+  createdAt?: Date;
+  updatedAt?: Date;
+  redeemedAt?: Date;
+  expiredAt?: Date;
+}
+
+export interface CustomFieldDefinition {
+  id?: string;
+  object?: string;
+  relatedType?: string;
+  name?: string;
+  userAccess?: string;
+  displayName?: string;
+  tooltip?: string;
+  createdAt?: Date;
+  updatedAt?: Date;
+  deletedAt?: Date;
+}
+
+export interface Plan {
+  id?: string;
+  object?: string;
+  code?: string;
+  state?: string;
+  name?: string;
+  description?: string;
+  intervalUnit?: string;
+  intervalLength?: number;
+  trialUnit?: string;
+  trialLength?: number;
+  totalBillingCycles?: number;
+  autoRenew?: boolean;
+  accountingCode?: string;
+  setupFeeAccountingCode?: string;
+  taxCode?: string;
+  taxExempt?: boolean;
+  currencies?: PlanPricing[];
+  hostedPages?: PlanHostedPages;
+  createdAt?: Date;
+  updatedAt?: Date;
+  deletedAt?: Date;
+}
+
+export interface PlanPricing {
+  currency?: string;
+  setupFee?: number;
+  unitAmount?: number;
+}
+
+export interface PlanHostedPages {
+  successUrl?: string;
+  cancelUrl?: string;
+  bypassConfirmation?: boolean;
+  displayQuantity?: boolean;
+}
+
+export interface AddOn {
+  id?: string;
+  object?: string;
+  planId?: string;
+  code?: string;
+  state?: string;
+  name?: string;
+  accountingCode?: string;
+  taxCode?: string;
+  displayQuantity?: boolean;
+  defaultQuantity?: number;
+  currencies?: AddOnPricing[];
+  createdAt?: Date;
+  updatedAt?: Date;
+  deletedAt?: Date;
+}
+
+export interface AddOnPricing {
+  currency?: string;
+  unitAmount?: number;
+}
+
+export interface ShippingMethod {
+  id?: string;
+  object?: string;
+  code?: string;
+  name?: string;
+  taxCode?: string;
+  createdAt?: Date;
+  updatedAt?: Date;
+  deletedAt?: Date;
+}
+
+
+
+export interface Empty {
+}
+
+export declare class Pager<T> {
+  count(): number;
+  first(): T;
+  each(): AsyncIterable<T>;
+  eachPage(): AsyncIterable<T>;
+}
+
+export declare class Client {
+  constructor(apiKey: string);
+  apiVersion(): string;
+  listSites(params: object): Pager<Site>;
+  getSite(siteId: string): Promise<Site>;
+  listAccounts(params: object): Pager<Account>;
+  createAccount(body: object): Promise<Account>;
+  getAccount(accountId: string): Promise<Account>;
+  updateAccount(accountId: string, body: object): Promise<Account>;
+  deactivateAccount(accountId: string): Promise<Account>;
+  getAccountAcquisition(accountId: string): Promise<AccountAcquisition>;
+  updateAccountAcquisition(accountId: string, body: object): Promise<AccountAcquisition>;
+  removeAccountAcquisition(accountId: string): Promise<Empty>;
```

Comment:
Do these methods actually return an empty object or are they `void`? If the latter, then `Promise<void>` will suffice.

Reply:
They return a special object of class Empty. This allows you to still get the request and response metadata. 

Example:

```
let r = await client.removeAccountAcquisition("id")
r.getResponse().rateLimitRemaining // 1985
```